pkg/registry/bundle.go

```diff
@@ -262,23 +290,36 @@ func (b *Bundle) cache() error {
 		}
 	}
 
-	if b.crds == nil {
-		b.crds = []*v1beta1.CustomResourceDefinition{}
-	}
 	for _, o := range b.Objects {
-		if o.GetObjectKind().GroupVersionKind().Kind == "CustomResourceDefinition" {
-			crd := &v1beta1.CustomResourceDefinition{}
-			// Marshal Unstructured and Decode as CustomResourceDefinition. FromUnstructured has issues
-			// converting JSON numbers to float64 for CRD minimum/maximum validation.
-			cb, err := o.MarshalJSON()
-			if err != nil {
-				return err
-			}
-			dec := serializer.NewCodecFactory(Scheme).UniversalDeserializer()
-			if _, _, err = dec.Decode(cb, nil, crd); err != nil {
-				return fmt.Errorf("error decoding CRD: %v", err)
+		if o.GroupVersionKind().Kind == CRDKind {
+			switch o.GroupVersionKind().Version {
+			case v1CRDVersion:
+				crd := &apiextensionsv1.CustomResourceDefinition{}
+				// Marshal Unstructured and Decode as CustomResourceDefinition. FromUnstructured has issues
+				// converting JSON numbers to float64 for CRD minimum/maximum validation.
+				cb, err := o.MarshalJSON()
+				if err != nil {
+					return err
+				}
+				dec := serializer.NewCodecFactory(Scheme).UniversalDeserializer()
+				if _, _, err = dec.Decode(cb, nil, crd); err != nil {
+					return fmt.Errorf("error decoding CRD: %v", err)
+				}
+				b.v1crds = append(b.v1crds, crd)
+			case v1beta1CRDVersion:
+				crd := &apiextensionsv1beta1.CustomResourceDefinition{}
+				// Marshal Unstructured and Decode as CustomResourceDefinition. FromUnstructured has issues
+				// converting JSON numbers to float64 for CRD minimum/maximum validation.
+				cb, err := o.MarshalJSON()
```

Comment:
nit: can we move the marshaling and universal deserialization above the case statement? seems like this code is identical